server/src/main/java/org/elasticsearch/cluster/metadata/DataStream.java

```diff
@@ -821,6 +877,10 @@ public void writeTo(StreamOutput out) throws IOException {
         if (out.getTransportVersion().onOrAfter(TransportVersions.V_8_500_020)) {
             out.writeOptionalWriteable(lifecycle);
         }
+        if (out.getTransportVersion().onOrAfter(DataStream.ADDED_FAILURE_STORE_TRANSPORT_VERSION)) {
+            out.writeString(TIMESTAMP_FIELD_NAME);
```

Comment:
Did you mean to serialize the timestamp again here? This seems like it was accidentally added since it doesn't match the `StreamInput` reading

Reply:
I thought the same, but check out readIndices() -- it reads (and throws away) a string to get around the fact that writeTo() writes an unused string. We probably ought to change readIndices() to not do that, and then could drop this out.writeString(). As something unrelated to this PR, we ought to stop writing that unused string in the first place (or at the very least make it an empty string instead of `@timestamp`).

Reply:
I think I'm going to go ahead and refactor that. I added this mostly for expediency, but it's certainly wasteful in this case.

Reply:
I removed the extra timestamp field name. Removing the first field usage likely will require a different PR. I wonder if this is a case of vestigial serialization sticking around from early data stream development that hasn't been cleaned up yet.